diff -pur orig/CypherParser.java fixed/CypherParser.java
--- orig/CypherParser.java	2016-09-14 19:52:07.735726512 +0200
+++ fixed/CypherParser.java	2016-09-14 19:57:00.904952589 +0200
@@ -557,7 +557,7 @@ public class CypherParser extends Parser
 		public WithContext with() {
 			return getRuleContext(WithContext.class,0);
 		}
-		public ReturnContext return() {
+		public ReturnContext return_() {
 			return getRuleContext(ReturnContext.class,0);
 		}
 		public ClauseContext(ParserRuleContext parent, int invokingState) {
@@ -647,7 +647,7 @@ public class CypherParser extends Parser
 				enterOuterAlt(_localctx, 9);
 				{
 				setState(213);
-				return();
+				return_();
 				}
 				break;
 			default:
@@ -1602,7 +1602,7 @@ public class CypherParser extends Parser
 		}
 	}
 
-	public final ReturnContext return() throws RecognitionException {
+	public final ReturnContext return_() throws RecognitionException {
 		ReturnContext _localctx = new ReturnContext(_ctx, getState());
 		enterRule(_localctx, 36, RULE_return);
 		try {
